### ISSUE TITLE
2.2.0-beta2: AssemblyRunner hangs when running in CoreCLR

### ISSUE THREAD

**Issue by daviwil** [xunit/xunit#960]
Hey guys,

Fair possibility that this could be user error, but I'm trying to write a simple harness that uses [AssemblyRunner](https://github.com/xunit/xunit/blob/master/src/xunit.runner.utility/Runners/AssemblyRunner.cs) and it seems to be hanging on test discovery when run in CoreCLR.  I've followed the example Brad provided on the [samples.xunit repo](https://github.com/xunit/samples.xunit/blob/190ac8fa0210f5c9c9d5fbb86680ca854241a40c/TestRunner/Program.cs). I've wired up all of the state and message functions and I don't see any of them being called (no output being printed).  I've tried the same code on .NET 4.5.1 and the tests seem to run fine.

Is there anything I should be aware of when trying to use AssemblyRunner in CoreCLR?  If you need a repro project I can whip one up this weekend.

Thanks!

**Comment by bradwilson** [comment]
Honestly, I've never tried that sample with .NET Core. I'll take a look this weekend and see if I can figure out what's going on, because there's nothing off the top of my head that suggests that it shouldn't work.

**Comment by daviwil** [comment]
I just created a simple repro project and everything works fine!  I'll have to investigate why my particular use case isn't working out.  I'll go ahead and close this issue.

Just in case any of this triggers an idea you might have, here's what I'm trying to accomplish.  I need to run some existing Xunit tests which use the PowerShell C# APIs, but since CoreCLR PowerShell has it's own custom AssemblyLoadContext and assembly management I need to run those unit tests *inside* of PowerShell.  Thus, I'm trying to create a simple test runner library project which I can load inside of PowerShell and then invoke its runner class on my test DLLs.

Since I can't 'dotnet run' my test runner DLL, I'm having to copy all of the Xunit DLL dependencies to be co-located with my runner DLL so that they can be found. I'm wondering if I'm just missing a dependency which would be getting loaded dynamically if it were found.  Strangely I'm not seeing any exceptions from the AssemblyRunner, it just never finishes discovering tests.

I'm going to try poking around a bit more in case I made a mistake somewhere :)
